test/filters/all.test.js

```diff
@@ -1,11 +1,73 @@
+import { hljs } from '@asyncapi/react-component';
 import { AsyncAPIDocument } from "@asyncapi/parser";
+
 import { 
+  includeFile,
+  retrieveLanguages,
+  loadLanguagesConfig,
   stringifySpec,
   stringifyConfiguration,
   renderSpec,
 } from "../../filters/all";
 
 describe('Filters', () => {
+  describe('.includeFile', () => {
+    it('should read content of file', async () => {
+      const content = includeFile('node_modules/@asyncapi/react-component/browser/standalone/without-parser.js');
+
+      expect(content instanceof Buffer).toEqual(true);
+      expect(content.toString().length > 0).toEqual(true);
+    });
+  });
+
+  describe('.retrieveLanguages', () => {
+    it('should retrieve list of languages', async () => {
+      const languages = retrieveLanguages('text\n```js\nfoo\n```\ntext abc\n```bash\nbar\n```\nxyz');
+
+      expect(languages.length).toEqual(2);
+      expect(languages).toEqual(['js', 'bash']);
+    });
+
+    it('should retrieve unique languages', async () => {
+      const languages = retrieveLanguages('text\n```js\nfoo\n```\ntext text\n```js\nfoo\n```\ntext');
+
+      expect(languages.length).toEqual(1);
+      expect(languages).toEqual(['js']);
+    });
+  });
+
+  describe('.loadLanguagesConfig', () => {
+    it('should load languages', async () => {
+      expect(hljs.getLanguage('csharp')).toEqual(undefined);
+      expect(hljs.getLanguage('ruby')).toEqual(undefined);
+
+      loadLanguagesConfig('text\n```csharp\nfoo\n```\ntext abc\n```ruby\nbar\n```\nxyz');
+
+      expect(hljs.getLanguage('csharp') === undefined).toEqual(false);
+      expect(hljs.getLanguage('ruby') === undefined).toEqual(false);
+    });
+
+    it('should load language using alias', async () => {
+      // use alias
+      expect(hljs.getLanguage('ts')).toEqual(undefined);
+
+      loadLanguagesConfig('abc\n```ts\nbar\n```\nxyz');
+
+      expect(hljs.getLanguage('ts') === undefined).toEqual(false);
+      expect(hljs.getLanguage('typescript') === hljs.getLanguage('ts')).toEqual(true);
+    });
+
+    it('should warn if at least one language does not exist', async () => {
+      const originalWarn = console.warn;
+      let warning;
+      console.warn = (...args) => { warning = args[0] };
+      loadLanguagesConfig('text\n```bash\nfoo\n```\ntext abc\n```foo_bar\nbar\n```\nxyz');
+      console.warn = originalWarn;
```

Comment:
You could mock the function here if you wanted, easier to restore the previous state as well.